cmd/format/format.go

```diff
@@ -205,33 +205,34 @@ func Run(v *viper.Viper, statz *stats.Stats, cmd *cobra.Command, paths []string)
 		return fmt.Errorf("failed to create walker: %w", err)
 	}
 
+	// start traversing
 	files := make([]*walk.File, BatchSize)
+
 	for {
+		// read the next batch
 		ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
-
 		n, err := reader.Read(ctx, files)
 
-		for idx := 0; idx < n; idx++ {
-			file := files[idx]
+		// ensure context is cancelled to release resources
+		cancel()
```

Comment:
For th erecord, I do not know enough about go to understand what this is about. I trust you, though =)

Reply:
You have to ensure `cancel()` is called on a context. Usually, it's done within a `defer`, but you can't do that within a `for` loop. `defer` is bound to a function.